modules/apps/document-library/document-library-preview-pdf/src/main/java/com/liferay/document/library/preview/pdf/internal/PDFPreviewableDLProcessor.java

```diff
@@ -473,11 +473,21 @@ else if (_documentConversion.isEnabled()) {
 						FileUtil.delete(file);
 					}
 
-					File file = _documentConversion.convert(
-						tempFileId, inputStream, extension, "pdf");
+					try {
+						File file = _documentConversion.convert(
+							tempFileId, inputStream, extension, "pdf");
 
-					_generateImages(
-						destinationFileVersion, file, maxNumberOfPages);
+						_generateImages(
+							destinationFileVersion, file, maxNumberOfPages);
+					}
+					catch (TimeoutException timeoutException) {
+						_fileVersionPreviewEventListener.onFailure(
+							destinationFileVersion);
+
+						if (_log.isDebugEnabled()) {
+							_log.debug(timeoutException);
+						}
+					}
```

Comment:
We shouldn't be propagating the timeout exn all over the place like this. Timeouts are an implementation detail. You should signal this with a regular io exception (wrapping the timeout exn).

Reply:
I will wrap it and put the failure on all the IOExceptions.

thanks!